ouroboros-consensus-test/test-consensus/Test/Consensus/MiniProtocol/BlockFetch/Client.hs

```diff
@@ -193,19 +231,52 @@ runBlockFetchTest BlockFetchClientTestSetup{..} = withRegistry \registry -> do
                       case Chain.applyChainUpdates updates chain of
                         Just chain' -> chain'
                         Nothing     -> error "Chain update failed"
-                -- Block until our "ChainSync" thread registered itself to the
-                -- FetchClientRegistry, see 'forkChainSync' below.
-                _ <- blockUntilJust $ Map.lookup peerId <$> readTVar varChains
+                    updateCPS cps =
+                      case CPS.applyChainUpdates (fmap TestHeader <$> updates) cps of
+                        Just cps' -> cps'
+                        Nothing   -> error "Chain update failed"
                 modifyTVar varChains $ Map.adjust updateChain peerId
+                modifyTVar (producerStateVars Map.! peerId) updateCPS
 
         forkChainSync peerId =
-          forkLinkedThread registry ("BracketSync" <> condense peerId) $
+            forkThread registry ("BracketSync" <> condense peerId) $
+            recordChainSyncExceptions $
             bracketSyncWithFetchClient fetchClientRegistry peerId $ do
-              let modifyChains = atomically . modifyTVar varChains
-              bracket_
-                (modifyChains $ Map.insert peerId Chain.Genesis)
-                (modifyChains $ Map.delete peerId)
-                (forkTicking peerId >>= waitThread)
+              let chainSyncTracer =
+                        (\ev -> show peerId <> ": ChainSyncClient: " <> show ev)
+                    >$< tracer
+                  chainDbView = (ChainSyncClient.defaultChainDbView chainDB) {
+                      ChainSyncClient.getCurrentChain =
+                        pure $ AF.Empty AF.AnchorGenesis
+                    }
+                  client =
+                    chainSyncClient
+                      (pipelineDecisionLowHighMark 10 20)
+                      chainSyncTracer
+                      topLevelConfig
+                      chainDbView
+                      ntnVersion
+                      (pure Mux.Continue)
```

Comment:
Does this mean the ChainSync client never terminates itself successfully?

... I would guess that the `bracketSyncWithFetchClient` terminates the ChainSync client (successfully?) when the BlockFetch client terminates itself successfully?